packages/form-validation/src/validation.js

```diff
@@ -252,7 +252,20 @@ export default class AuroFormValidation {
        * The validityState definitions are located at https://developer.mozilla.org/en-US/docs/Web/API/ValidityState.
        */
 
-      let hasValue = elem.value && elem.value.length > 0;
+      let hasValue = false;
+
+      // Check string for having a value
+      if (typeof elem.value === "string") {
+        hasValue = elem.value && elem.value.length > 0;
+      }
+
+      // Check array value types for having a value
+      if (Array.isArray(elem.value)) {
+        hasValue = Boolean(
+          elem.value.length > 0 &&
+          elem.value.some((value) => typeof value === "string" && value.length > 0)
+        )
```

Comment:
```suggestion
        hasValue = elem.value.some((value) => typeof value === "string" && !!value);
```
This is sufficient as false will get returned when the array has no item.

Reply:
I understand open/closed principal but how is this not extendable?

we can extend like this.
```
hasValue = 
  elem.value.some((value) => typeof value === "string" && value.length > 0) &&
  elem.value.filter((value) => value.length > 3)?.length // at least one value is longer than 3
```

Reply:
I'm not sure I'm understanding what your argument is.

Why shouldn't we have an explicit and adjustable threshold check?

```javascript
Boolean(
  elem.value.length > 0 && // This line lets us make sure there's N values, which could be adjustable later without modification to the code
  elem.value.some((value) => typeof value === "string" && value.length > 0) // Other checks run as normal
)
```
If later we wanted to make a threshold value for menu's required so that it required at minimum two selections, we could just change that line to something like:
```javascript
Boolean(
  elem.value.length > elem.minSelectedValues &&
  elem.value.every((value) => typeof value === "string" && value.length > 0)
)
```
This helps make future changes easier, commits more readable, etc.